### ISSUE TITLE
False positive in MISRA C++ 2008 Rule 6-5-6

### ISSUE THREAD

**Issue by d3lta-v** [naivesystems/analyze#10]
**Describe the bug**
The bug is found in the checker for MISRA C++ 2008 Rule 6–5-6, which states that loop control variables in a for loop, other than the loop counter itself, should have the type `bool` to make the program easier to read. The analyzer seems to apply this rule to any loop control variable even if they are not changed inside the loop.

For example, the following snippet of code compares the loop counter `i` with a `const size_t` variable, which is guaranteed to never change: 
```c++
void SerPort::ReadCallback(const uint8_t* buf, const size_t len) {
    for (size_t i=0; i<len; i++)
    ...
```

This is flagged by Analyze as a rule 6-5-6 violation, possibly due to the usage of the variable `len` within the loop condition, even though this is a `const` variable.

**To Reproduce**
1. Analyze any code with a for loop that has a variable in the loop condition as stated above
2. The analyzer software will report this as a 6-5-6 violation

**Expected behavior**
The violation should not be flagged by the software.

**Desktop (please complete the following information):**
 - OS: Fedora 40 running Analyze through Podman 
 - Version: latest

**Comment by Peytonwilson223** [comment]
Your walkthrough of the false positive clearly shows how the analyzer indiscriminately flags any variable used in the for‑loop condition, even when a const size_t like len never changes, which directly contradicts MISRA C++ 2008 Rule 6‑5‑6’s intent. In practice, tracing such anomalies feels similar to consulting https://macombcountycourts.org to verify whether a precedent truly applies. The expected behavior—ignoring immutable control variables—would keep the diagnostic signal meaningful and reduce noise. Hopefully the next release refines the rule engine to respect constness in loop conditions.

**Comment by AdamVibe19** [comment]
The false positive you observed stems from the analyzer treating the const size_t len in the for‑condition as a mutable loop control variable, which violates the spirit of MISRA C++ 2008 Rule 6‑5‑6. Such over‑generalization mirrors how a generic query might pull irrelevant https://marioncountycourts.org when the specific case context is ignored. A targeted improvement would be to add a const‑ness check before flagging, ensuring only variables that can change trigger the rule. Doing so should suppress the spurious warning and keep the analysis aligned with the intended readability guidelines.

**Comment by bhayiali234** [comment]
I found the discussion about the MISRA C++ 2008 Rule 6-5-6 false positive quite interesting, especially the challenges of distinguishing genuine issues from analyzer behavior. Clear examples and context can make these cases much easier to assess. I also find resources like [Martin Appraisal Information](https://martincountypropertyappraiser.org) useful when researching structured information. It would be helpful to see how others have handled similar false positives.
